presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/JoinMatcher.java

```diff
@@ -29,11 +29,18 @@
 {
     private final JoinNode.Type joinType;
     private final List<AliasPair> equiCriteria;
+    private final JoinNode.DistributionType distributionType;
 
     JoinMatcher(JoinNode.Type joinType, List<AliasPair> equiCriteria)
     {
+        this(joinType, equiCriteria, null);
+    }
+
+    JoinMatcher(JoinNode.Type joinType, List<AliasPair> equiCriteria, JoinNode.DistributionType distributionType)
+    {
         this.joinType = requireNonNull(joinType, "joinType is null");
         this.equiCriteria = requireNonNull(equiCriteria, "equiCriteria is null");
+        this.distributionType = distributionType;
```

Comment:
Actually I think that for each field we want to test in matcher we should have a:
`Predicate<FieldType>` which we blindly test in `matches` method. We would set it to `Predicates.true()` for any field we do not care about.

cc: @kokosing